domain_access/src/AccessMethod/NoRecords.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace Drupal\domain_access\AccessMethod;
+
+use Drupal\Core\Session\AccountInterface;
+use Drupal\domain_access\DomainAccessMethodInterface;
+use Drupal\node\NodeInterface;
+
+/**
+ * Class Null.
+ *
+ * @package Drupal\domain_access
+ */
+class NoRecords implements DomainAccessMethodInterface {
```

Comment:
I don't think it's our responsibility to provide the NULL service. You'd have to alter the service container to activate this anyway.

Overall, I think the cleaner solution is to answer the question "What front-end features do you need that aren't available without Domain Access?"

Reply:
You can create a service provider https://cgit.drupalcode.org/domain_adv/tree/src/DomainAdvServiceProvider.php to make this change on the fly. 

I can remove the NoRecords class and have other modules provide their own implementation.

The Domain access module provides all the content domain assignment code, so you can specify where domains the content is available on. With Domain access advanced I need all the records and everything, I just need the node access records to not be there.

I did investigated the use of the `hook_node_grants_alter()` and `hook_node_access_records_alter()` but unlike the old `hook_domainrecords()` and `hook_domaingrants()` I could just `return array();` and not worry about records from other modules implementing node access.